### ISSUE TITLE
Fluent Builder API vs Builder pattern

### ISSUE THREAD

**Issue by jansupol** [jakartaee/config#210]
The current Loader API looks
```
Loader.bootstrap().load()
```

However, we probably want to add other enrichments to the API, such as converters or sources. Then we would have something like
```
Loader.bootstrap().converter(...).source(...).load()
```
Which would make Loader mutable. Then we have two possibilities:

1. To return a new instance of Loader for each converter, source, and other calls
2. To use a Builder pattern using `Loader.bootstrap().build().load()` which would more nicely turn into `Loader.builder().build().load()`

----
In summary, should we use `bootstrap()` method, or `builder()` method?

**Comment by hantsy** [comment]
For the conversion/source etc. configuration element, I hope to adapt the Java 8 fluent APIs (`Function/Consumer/Supplier`).

The following is HTTP Security config in Spring Security 6. 

```java
http
  .authorizeHttpRequests(authorize -> authorize
  	.anyRequest().authenticated()
  )
  .formLogin(...)
  .httpBasic(...);
```

We should adopt the `code style of the new Java Language feature` here... no relation to the Security API.
